docs/source/functions.rst

```diff
@@ -0,0 +1,430 @@
+snowflake.snowpark.functions
+================================
+Provides utility and SQL functions that generate :class:`~snowflake.snowpark.Column` expressions that you can pass to :class:`~snowflake.snowpark.DataFrame` transformation methods.
```

Comment:
When using `automodule`, this docstring will be extracted automatically from `src/snowflake/snowpark/functions.py`, I have yet to figure out how to do that using `currentmodule`.

Reply:
So you mean text in this file has to be manually extracted from src/snowflake/snowpark/functions.py? Hmm it might not be a good idea since it's hard to keep them the same if someone has a change in src/snowflake/snowpark/functions.py?

Reply:
Yes, I don't have a better workaround for now. What if we just move the text to be in the documentation only?

Reply:
TODO: Let's have a JIRA for this.